pkg/operator/cockroachdb/controller.go

```diff
@@ -187,8 +187,11 @@ func (c *ClusterController) onUpdate(oldObj, newObj interface{}) {
 }
 
 func (c *ClusterController) onDelete(obj interface{}) {
-	cluster := obj.(*cockroachdbv1alpha1.Cluster).DeepCopy()
-	logger.Infof("cluster %s deleted from namespace %s", cluster.Name, cluster.Namespace)
+	cluster, ok := obj.(*cockroachdbv1alpha1.Cluster)
+	if ok {
```

Comment:
@phantooom When will this type assertion fail? I haven't heard a report of this issue before. When have you seen it crash?

Reply:
onDelete func in some case will recive a DeletedFinalStateUnknown obj, when it retrun DeletedFinalStateUnknown  obj,it will panic

doc https://github.com/kubernetes/client-go/blob/12a0647ebb467f3bbc9ed7ce978bc6b1be26b9d1/tools/cache/controller.go#L173

@travisn